### PR TITLE
add more constants to tarfile

### PR DESCRIPTION
**Pull request by FlorianLudwig**
This adds more constants from `tarfile` and reorders them to match the source file.

### CODE REVIEW

File: stdlib/2and3/tarfile.pyi [hunk 1]
```diff
@@ -15,23 +15,53 @@ elif sys.version_info >= (3,):
...
 DIRTYPE: bytes
 FIFOTYPE: bytes
-CONTTYPE: bytes
 CHRTYPE: bytes
-BLKTYPE: bytes
-GNUTYPE_SPARSE: bytes
+
+GNUTYPE_LONGNAME: bytes
+GNUTYPE_LONGLINK: bytes
+GNUTYPE_SPARSE: byte
```
Comment (hauntsaninja) [inline stdlib/2and3/tarfile.pyi:41]: typo!

Comment (FlorianLudwig) [reply stdlib/2and3/tarfile.pyi]: Thanks, fixed!

**Comment by hauntsaninja** [comment]
`Tuple[T]` is a single element tuple with that element of type `T`. You'll want `Tuple[T, ...]` for wherever CI is complaining.

File: stdlib/2and3/tarfile.pyi [hunk 2]
```diff
@@ -15,24 +15,54 @@ elif sys.version_info >= (3,):
...
+GNU_FORMAT: int
+PAX_FORMAT: int
+DEFAULT_FORMAT: int
+
+# tarfile constants
+
+SUPPORTED_TYPES: Tuple[bytes]
+REGULAR_TYPES: Tuple[bytes]
+GNU_TYPES: Tuple[bytes]
+PAX_FIELDS: Tuple[str]
```
Comment (srittau) [inline stdlib/2and3/tarfile.pyi:57]: `Tuple[bytes, ...]` etc. Otherwise they would be tuples of length 1.

File: stdlib/2and3/tarfile.pyi [hunk 3]
```diff
@@ -15,24 +15,54 @@ elif sys.version_info >= (3,):
...
+PAX_NUMBER_FIELDS: Dict[str, type]
+
+if sys.version_info >= (3,):
+    PAX_NAME_FIELDS: set
```
Comment (srittau) [inline stdlib/2and3/tarfile.pyi:61]:
```suggestion
    PAX_NAME_FIELDS: Set[str]
```

**Pull request merged**